thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalyTimeBasedSummarizer.java

```diff
@@ -132,5 +140,35 @@ private static void populateMergedResult(MergedAnomalyResultDTO mergedAnomaly,
     mergedAnomaly.setStartTime(currentResult.getStartTime());
     mergedAnomaly.setEndTime(currentResult.getEndTime());
     mergedAnomaly.setCreatedTime(System.currentTimeMillis());
+    // populate current result's property as well, will be used to identify if two anomalies can be merged by comparing their mergeable keys
+    mergedAnomaly.setProperties(AnomalyDetectionUtils.decodeCompactedPropertyStringToMap(currentResult.getProperties()));
+  }
+
+
+  /**
+   * Given property keys from anomaly function, comparing if two anomalies have same property on the mergeable keys when doing anomaly detection
+   * If key set is empty, or both properties for the two anomalies are empty or if all of the values on mergeable keys are equal on anomalies return true
+   * Otherwise return false
+   * @param anomaly1 The first anomaly result
+   * @param anomaly2 The second anomaly result
+   * @param mergeableKeys keys that passed by AnomalyMergeConfig, which is defined by Anomaly Detection Function
+   * @return true if two anomalies are equal on mergeable keys, otherwise return false
+   */
+  private static boolean isEqualOnMergeableKeys(MergedAnomalyResultDTO anomaly1, MergedAnomalyResultDTO anomaly2, List<String> mergeableKeys){
+    Map<String, String> prop1 = anomaly1.getProperties();
+    Map<String, String> prop2 = anomaly2.getProperties();
+    // degenerate case
+    if(mergeableKeys.size() == 0 ||
+        (prop1.isEmpty() && prop2.isEmpty())){
```

Comment:
This assumption might not always hold. It could be accidentally changed to null at somewhere.